app/middlewares/session_validation.js

```diff
@@ -0,0 +1,26 @@
+const errors = require('../errors');
+const jwt = require('jwt-simple');
+const {
+  common: {
+    session: { secret }
+  }
+} = require('../../config');
+const { findUser } = require('../services/users');
+
+exports.checkSession = (req, _, next) => {
+  const token = req.headers.authorization;
+  try {
```

Comment:
when you use try and catch, you have to use async await approach
here is a good idea to use it since the promise is inside an if